File: CSharp/BackingCode/FlightRepository.cs
namespace BackingCode
{

    using System;
    using System.Collections.Generic;
    using System.Linq;

    public class FlightRepository
    {
        private List<FlightDto> flights = new List<FlightDto>();

        public void Save(FlightDto flight) => flights.Add(flight);
        public IEnumerable<FlightDto> FindAll() => flights.AsReadOnly();
        public IEnumerable<FlightDto> FindByOriginAirportCode(String airportCode) => flights.Where(it => it.OriginAirportCode == airportCode);
    }
}

File: src/main/java/ch/scs/cc/testing/smells/FlightRepository.java
package ch.scs.cc.testing.smells;

import java.util.ArrayList;
import java.util.List;

import static java.util.stream.Collectors.toList;

public class FlightRepository {

    private List<FlightDto> flights = new ArrayList<>();

    public void save(FlightDto flight) {
        flights.add(flight);
    }

    public List<FlightDto> findByOriginAirportCode(String airportCode) {
        return flights.stream()
                .filter(it -> it.originAirportCode().equals(airportCode))
                .collect(toList());
    }
}
